model-optimizer/mo/utils/ir_reader/layer_to_class.py

```diff
@@ -280,6 +280,42 @@ def copy_input_blobs(op: Node, copy_op: Node):
     'TopK': TopKNormalizer.normalize_outputs,
 }
 
+def restore_and_provide_tensor_names(op: Node):
+    debug_dict = {}
+
+    for out_port in op.ports:
+        # op.ports is our internal attribute, dictionary, where keys are numbers of output ports
+        # and values are tuples with shape and tensor name:
+        # {out_port_idx_1: (out_port_idx_1_shape, out_port_idx_1_tensor_name),
+        #  out_port_idx_1: (out_port_idx_1_shape, out_port_idx_1_tensor_name)}
+        out_tensor_names = op.ports[out_port][1]
+
+        # handle Constant operations with old style output port numbering
+        if op.soft_get('type') == 'Const':
+            assert len(op.ports) == 1, 'Something wrong with Constant node: {}, wrong number ' \
+                                       'of output ports: {}!'.format(op.soft_get('name'), len(op.ports))
+            out_port = 0
+
+        debug_dict[out_port] = []
+        if out_tensor_names is not None:
+            # handle tensor names with commas and add them to dictionary as separated
+            if out_tensor_names.find(',') >= 0:
+                str_to_replace = '<comma_in_tensor_name>'
+                out_tensor_names = (out_tensor_names.replace('\\,', str_to_replace)).split(',')
+                for out_tensor_name in out_tensor_names:
+                    debug_dict[out_port].append((out_tensor_name, out_port,
+                                                 out_tensor_name.replace(str_to_replace, ',')))
+            else:
+                debug_dict[out_port] = [(out_tensor_names, out_port, out_tensor_names)]
+        else:
+            debug_dict[out_port] = None
+
+    # provide tensor names to data nodes
+    if len(debug_dict) > 0:
+        for out in op.out_nodes():
+            if debug_dict[out + len(op.in_nodes())] is not None:
+                op.out_node(out)['fw_tensor_debug_info'] = debug_dict[out + len(op.in_nodes())]
```

Comment:
We do we need a separate `if` to do this? Let's update the data nodes in the previous for loop. There is no need in `debug_dict` also. Please, refactor the code.

Reply:
Yes, you are right. This code was refactored.